src/dodal/beamlines/i18.py

```diff
@@ -0,0 +1,266 @@
+        fake_with_ophyd_sim,
+    )
+
+
+# not ready yet
+@skip_device()
+def undulator(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> Undulator:
+    return device_instantiation(
+        Undulator,
+        "undulator",
+        f"{BeamlinePrefix(BL).insertion_prefix}-MO-SERVC-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        bl_prefix=False,
+        poles=None,  # todo check the real values
+        length=None,  # todo check the real values
+    )
+
+
+@skip_device()
+def slits_1(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> Slits:
+    return numbered_slits(
+        1,
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+# Must document what PandAs are physically connected to
+# See: https://github.com/bluesky/ophyd-async/issues/284
+@skip_device()
+def panda1(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> HDFPanda:
+    return device_instantiation(
+        HDFPanda,
+        "panda1",
+        "-MO-PANDA-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        path_provider=get_path_provider(),
+    )
+
+
+# odin detectors are not yet supported.
+# There is a controls project in the works,
+# not ready anytime soon
+@skip_device()
+def xspress3_odin(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> Xspress3:
+    return device_instantiation(
+        Xspress3,
+        prefix="-EA-XSP-02:",
+        name="xspress3_odin",
+        num_channels=4,
+        wait=wait_for_connection,
+        fake=fake_with_ophyd_sim,
+    )
+
+
+@skip_device
+def dcm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> DoubleCrystalMonochromator:
+    crystal_1_metadata = CrystalMetadata(
+        usage="Bragg",
+        type="silicon",
+        reflection=(1, 1, 1),
+        d_spacing=(3.13475, "nm"),
+    )
+
+    crystal_2_metadata = CrystalMetadata(
+        usage="Bragg",
+        type="silicon",
+        reflection=(1, 1, 1),
+        d_spacing=(3.13475, "nm"),
+    )
+    # todo double check if crystal metdata is correct
+
+    return device_instantiation(
+        DoubleCrystalMonochromator,
+        "dcm",
+        f"{BeamlinePrefix(BL).beamline_prefix}-MO-DCM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        bl_prefix=False,
+        temperature_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-DI-DCM-01:",
+        crystal_1_metadata=crystal_1_metadata,
+        crystal_2_metadata=crystal_2_metadata,
+    )
+
+
+@skip_device()
+def i0(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "i0",
+        "-DI-XBPM-02:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="Cividec Diamond XBPM",
+        path_provider=get_path_provider(),
+    )
```

Comment:
```suggestion
def i0(
    wait_for_connection: bool = True,
    fake_with_ophyd_sim: bool = False,
) -> TetrammDetector:
    return device_instantiation(
        TetrammDetector,
        "i0",
        "-DI-XBPM-02:",
        wait_for_connection,
        fake_with_ophyd_sim,
        path_provider=get_path_provider(),
    )
```

Should: this can be unskipped.
nit: alternative form:

```python
@device_factory()
def i0() -> TetrammDetector:
    return TetrammDetector(
        f"{PREFIX.beamline_prefix}-DI-XBPM-02:",
        path_provider=get_path_provider(),
    )
```